src/renderers/dom/shared/__tests__/DOMPropertyOperations-test.js

```diff
@@ -261,4 +261,20 @@ describe('DOMPropertyOperations', () => {
       );
     });
   });
+
+  describe('getValueForProperty', function() {
+    it('uses getValueForAttribute when given a custom attribute', function() {
+      var node = document.createElement('div');
+
+      node.setAttribute('data-test', 'test');
+
+      var value = DOMPropertyOperations.getValueForProperty(
```

Comment:
I can give it a shot! I'm curious if we can even approach this code path externally. I only discovered it after I removed the custom attributes checks for data and aria attributes.